src/GraphQL.MicrosoftDI.Tests/FieldBuilderExtensionTests.cs

```diff
@@ -0,0 +1,398 @@
+            VerifyScoped();
+        }
+
+        [Fact]
+        public void WithScope4()
+        {
+            var graph = new ObjectGraphType();
+            var field = graph.Field<StringGraphType>()
+                .Resolve()
+                .WithScope()
+                .WithService<string>()
+                .WithService<int>()
+                .WithService<short>()
+                .WithService<byte>()
+                .Resolve((context, value, v2, v3, v4) => value + v2 + v3 + v4)
+                .FieldType;
+            field.Resolver.Resolve(_scopedContext).ShouldBe("hello234");
+            VerifyScoped();
+        }
+
+        [Fact]
+        public void WithScope5()
+        {
+            var graph = new ObjectGraphType();
+            var field = graph.Field<StringGraphType>()
+                .Resolve()
+                .WithScope()
+                .WithService<string>()
+                .WithService<int>()
+                .WithService<short>()
+                .WithService<byte>()
+                .WithService<long>()
+                .Resolve((context, value, v2, v3, v4, v5) => value + v2 + v3 + v4 + v5)
+                .FieldType;
+            field.Resolver.Resolve(_scopedContext).ShouldBe("hello2345");
+            VerifyScoped();
+        }
+
+        [Fact]
+        public void WithScope2Alt()
+        {
+            var graph = new ObjectGraphType();
+            var field = graph.Field<StringGraphType>()
+                .Resolve()
+                .WithScope()
+                .WithService<string, int>()
+                .Resolve((context, value, v2) => value + v2)
+                .FieldType;
+            field.Resolver.Resolve(_scopedContext).ShouldBe("hello2");
+            VerifyScoped();
+        }
+
+        [Fact]
+        public void WithScope3Alt()
+        {
+            var graph = new ObjectGraphType();
+            var field = graph.Field<StringGraphType>()
+                .Resolve()
+                .WithScope()
+                .WithService<string, int, short>()
+                .Resolve((context, value, v2, v3) => value + v2 + v3)
+                .FieldType;
+            field.Resolver.Resolve(_scopedContext).ShouldBe("hello23");
+            VerifyScoped();
+        }
+
+        [Fact]
+        public void WithScope4Alt()
+        {
+            var graph = new ObjectGraphType();
+            var field = graph.Field<StringGraphType>()
+                .Resolve()
+                .WithScope()
+                .WithService<string, int, short, byte>()
+                .Resolve((context, value, v2, v3, v4) => value + v2 + v3 + v4)
+                .FieldType;
+            field.Resolver.Resolve(_scopedContext).ShouldBe("hello234");
+            VerifyScoped();
+        }
+
+        [Fact]
+        public void WithScope5Alt()
+        {
+            var graph = new ObjectGraphType();
+            var field = graph.Field<StringGraphType>()
+                .Resolve()
```

Comment:
Looking at these calls, I cannot get rid of the thought that the `.Resolve()` method is redundant. Of course it serves as the entry point for fluent APIs on top of created builder. Nevertheless, one could do without it if we provide several entry points for `WithScope` and `WithService` methods. It takes a little more code, but it's pretty simple.

Reply:
You can but then it is not clear where you are entering the other builder. Because you can’t write WithScope().WithArgument(...)

Reply:
Having Resolve at the start matches the syntax of Field where Field with no arguments is a builder. Having Resolve at the end concludes it with the delegate.

Reply:
Actually you can’t. See below comment.

Reply:
> Having Resolve at the start matches the syntax of Field where Field with no arguments is a builder. Having Resolve at the end concludes it with the delegate.

Got it. By the way why this `Field` method is an instance one and not extension like newly added?

Reply:
I agree that `Field()` should be an extension, not an instance method.